src/docfx/restore/RestoreUrl.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    public static class RestoreUrl
+    {
+        private const int MaxVersionCount = 5;
+
+        public static string GetRestoreVersionPath(string restoreDir, string version)
+            => PathUtility.NormalizeFile(Path.Combine(restoreDir, version));
+
+        public static string GetRestoreRootDir(string address)
+            => Docs.Build.Restore.GetRestoreRootDir(address, AppData.UrlRestoreDir);
+
+        public static async Task<string> Restore(string docset, string address)
+        {
+            var tempFile = await DownloadToTempFile(address);
+
+            var fileVersion = "";
+            using (var fileStream = File.Open(tempFile, FileMode.Open, FileAccess.Read))
+            {
+                fileVersion = PathUtility.Encode(HashUtility.GetMd5String(fileStream));
```

Comment:
>GetMd5String [](start = 61, length = 12)

suggest to use sha1 or sha2. md5 is only used to generate document id for backward compatibility purposes.

Reply:
will sha1 and sha2 be faster than md5?

---
In reply to: [202255164](https://github.com/dotnet/docfx/pull/3065#discussion_r202255164) [](ancestors = 202255164)